voxblox_ros/src/visualize_tsdf.cc

```diff
@@ -0,0 +1,205 @@
+#include <string>
+
+#include <gflags/gflags.h>
+#include <glog/logging.h>
+#include <pcl/conversions.h>
+#include <pcl/io/ply_io.h>
+#include <pcl/point_types.h>
+#include <pcl_conversions/pcl_conversions.h>
+#include <pcl_msgs/PolygonMesh.h>
+#include <pcl_ros/point_cloud.h>
+#include <ros/ros.h>
+#include <sensor_msgs/PointCloud2.h>
+
+#include <voxblox/core/tsdf_map.h>
+#include <voxblox/io/layer_io.h>
+#include <voxblox/io/mesh_ply.h>
+#include <voxblox/mesh/mesh_integrator.h>
+
+#include "voxblox_ros/mesh_pcl.h"
+#include "voxblox_ros/mesh_vis.h"
+#include "voxblox_ros/ptcloud_vis.h"
+
+DEFINE_string(tsdf_proto_path, "", "Path to the serialized TSDF grid.");
+
+DEFINE_double(tsdf_surface_distance_threshold_factor, 0.75,
+              "This factor multiplied with the voxel size determines the "
+              "maximum distance to the surface for voxels to be considered "
+              "near-surface voxels.");
+
+DEFINE_string(tsdf_world_frame, "world",
+              "Name of the world frame. This frame will be used to publish all "
+              "the pointclouds and meshes.");
+
+DEFINE_string(tsdf_mesh_color_mode, "color",
+              "Color mode for the TSDF mesh extraction, options: "
+              "[color, height, normals, lambert, gray]");
+
+DEFINE_string(tsdf_voxel_ply_output_path, "",
```

Comment:
Why gflags???? ROS params are so much easier. This is anyway a ROS node that you launch with ROS run.
Unless there's a good reason for gflags, I *strongly* prefer ROS params especially since the rest of the package only uses ROS params.